### PR TITLE
DEV-172 hexcache

### PR DESCRIPTION
**Pull request by taybenlor**

### CODE REVIEW

**Review by wryun — APPROVED** [review]
Looks appropriately terrible to me.

File: buildspec.yaml [hunk 1]
```diff
@@ -15,6 +15,11 @@ phases:
       - DEPLOY_HASH="$(git rev-parse HEAD)"
       # https://makecode.com/cli/staticpkg
       - node node_modules/pxt-core/built/pxt.js staticpkg -r "$DEPLOY_HASH"
+      # We store the built hexfiles in our repo, then copy them into the cache
+      # This prevents us needing the compilation api to build
+      - cp built_hexfiles/* built/packaged/$DEPLOY_HASH/hexcache/
```
Comment (wryun) [inline buildspec.yaml]: I ended up fixing the source last time:

8d22dc03f02cdbcbd8bdb067f6a3e5d510bb8bbd

which we could do this time as well (for connect.png). But it seems like there's some fundamental mess going on here, so :man_shrugging:

Comment (wryun) [reply buildspec.yaml]: Oh yeah, also, can you do the symlink version instead of the cp? (see what that commit killed)

**Review by wryun — CHANGES_REQUESTED** [review]

File: package.json [hunk 2]
```diff
@@ -11,6 +11,12 @@
     "type": "git",
     "url": "git+https://github.com/Microsoft/pxt-microbit.git"
   },
+  "scripts": {
+    "start": "npm run build && npm run serve",
```
Comment (wryun) [inline package.json:15]: might be worth discussing the trade-offs in readme? i.e. that we're not using pxt serve, so we don't get hot-reloading (or the API), but we also avoid issues with differing between dev and prod.

File: buildspec.yaml (hunk 1 above)
Comment (wryun) [reply buildspec.yaml]: Why not use the npm scripts?

**Comment by taybenlor** [comment]
@wryun back to you!

File: package.json (hunk 2 above)
Comment (wryun) [reply package.json:15]: Nit: think you might have missed this one. Not important.

**Review by wryun — APPROVED** [review]

File: package.json (hunk 2 above)
Comment (taybenlor) [reply package.json:15]: > This task needs to be run each time you make a change. Which is slow, but is necessary so that we emulate prod.

Not enough?

Comment (wryun) [reply package.json:15]: Ah, yeah, good, but I would have also mentioned that the 'normal' way of doing this is to use 'pxt serve', but we've found it frequently inconsistent with what staticpkg produces. i.e. if someone wants to get adventurous in the future, warn them about the landmine.

**Pull request merged**